dev-libs/quazip/quazip-0.7.2.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils flag-o-matic multibuild qmake-utils
+
+DESCRIPTION="A simple C++ wrapper over Gilles Vollant's ZIP/UNZIP package"
+HOMEPAGE="http://quazip.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ppc ~ppc64 ~x86 ~x86-fbsd ~amd64-linux ~x86-linux"
+IUSE="qt4 +qt5 static-libs test"
+
+REQUIRED_USE="|| ( qt4 qt5 )"
+
+RDEPEND="
+	sys-libs/zlib[minizip]
+	qt4? ( dev-qt/qtcore:4 )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtnetwork:5
+	)
+"
+DEPEND="${RDEPEND}
+	test? (
+		qt4? ( dev-qt/qttest:4 )
+	)
+"
+
+DOCS=( NEWS.txt README.txt )
+HTML_DOCS=( doc/html/. )
+
+MULTIBUILD_VARIANTS=( qt4 qt5 )
+
+pkg_setup() {
+	MULTIBUILD_VARIANTS=( $(usev qt4) $(usev qt5) )
+}
+
+src_prepare() {
+	use static-libs || \
+		sed -e "/^install/ s/quazip_static//" -i quazip/CMakeLists.txt || die
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	myconfigure() {
+		local libdir=$(get_libdir)
+		local mycmakeargs=(
+			-DLIB_SUFFIX=${libdir/lib/}
+		)
+		unset libdir
+		if [[ ${MULTIBUILD_VARIANT} = qt4 ]]; then
+			mycmakeargs+=( -DBUILD_WITH_QT4=ON )
+		fi
+		if [[ ${MULTIBUILD_VARIANT} = qt5 ]]; then
+			append-cxxflags -std=c++11 -fPIC
```

Comment:
this is dangerous, because `CXXFLAGS` is still global. Before `append-cxxflags`, make `CXXFLAGS` local:
`local -x CXXFLAGS="${CXXFLAGS}"`